crates/iceberg/src/scan.rs

```diff
@@ -52,6 +62,18 @@ impl<'a> TableScanBuilder<'a> {
         self
     }
 
+    /// Sets the scan as being case-insensitive
+    pub fn with_case_insensitivity(mut self) -> Self {
```

Comment:
This API is a bit strange to me. How about using `with_case_sensitive(v: bool)` for clear?

Reply:
I'm open to that. I assumed that being case-sensitive would be the default, and so you'd only need to call the builder method if you wanted the opposite of that. Keen to hear other thoughts.